src/api.js

```diff
@@ -11,4 +11,13 @@ const api = axios.create({
       .catch(error => {
         throw new Error(`Error fetching articles: ${error.response?.data?.message || error.message}`);
       });
+  };
+
+  export const fetchSingleArticle = (article_id) => {
+    return api
+      .get(`/articles/${article_id}`)
+      .then(({ data }) => data)
+      .catch((error) => {
+        throw new Error(`Error fetching article: ${error.response?.data?.message || error.message}`);
```

Comment:
why are we throwing our a new error here ? cant we use the existing error to do the handling ?